org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/AbstractServiceReferenceAssert.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.servicereference;
+
+import static org.osgi.test.assertj.bundle.BundleAssert.BUNDLE;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import org.assertj.core.api.AbstractComparableAssert;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.assertj.core.api.ListAssert;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.Version;
+import org.osgi.test.assertj.bundle.BundleAssert;
+import org.osgi.test.assertj.dictionary.DictionaryAssert;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+public abstract class AbstractServiceReferenceAssert<SELF extends AbstractServiceReferenceAssert<SELF, ACTUAL>, ACTUAL extends ServiceReference<?>>
+	extends AbstractComparableAssert<SELF, ACTUAL> {
+
+	protected AbstractServiceReferenceAssert(ACTUAL actual, Class<?> selfType) {
+		super(actual, selfType);
+	}
+
+	public DictionaryAssert<String, Object> hasServicePropertiesThat() {
+		return new DictionaryAssert<String, Object>(Dictionaries.asDictionary(actual));
+	}
+
+	public BundleAssert hasBundleThat() {
```

Comment:
I think this would read better as `isRegisteredInBundleThat()` and ties in better with the other assertion methods.